### PR TITLE
Feature/spec posts controller

### PR DESCRIPTION
**Pull request by dpienczyn**

### CODE REVIEW

File: spec/controllers/posts_controller_spec.rb [hunk 1]
```diff
@@ -20,16 +23,122 @@
...
+        put :update, params: { id: post.id, post: invalid_params }
+      end
+
+      it "will render edit template" do
+        sign_in(user)
+        subject
+        expect(response).to render_template(:edit)
+      end
+    end
   end
```
Comment (mosinski) [inline spec/controllers/posts_controller_spec.rb]:
```
   describe "PUT #update" do
     let(:other_user) { create(:user) }
     let(:post) { create(:post, user: user) }
     let(:post_params) { attributes_for(:post, title: 'test') }

     subject do
       put :update, params: { id: post.id, post: post_params }
     end

     context 'when user is an author' do
       it 'updates requested post' do
         sign_in(user)
         subject
         post.reload
         expect(post.title).to eq('test')
        end
      end

     context 'when user is not an author' do
       it 'not updates requested post' do
         sign_in(other_user)
         subject
         expect(response).to_not be_successful
       end
     end

     context 'when invalid params' do
       let(:post_params) { attributes_for(:post, title: '') }

       it "will render edit template" do
         sign_in(user)
         subject
         expect(response).to render_template(:edit)
       end
     end
   end
```

File: spec/controllers/posts_controller_spec.rb [hunk 2]
```diff
@@ -20,16 +23,122 @@
...
+      before {
+        sign_in(user)
+        post :create, params: { post: { title: "", description: "lololol", author: "ola" } }
+      }
+
+      it "will render new template" do
+        expect(response).to render_template(:new)
+      end
+    end
+  end
```
Comment (mosinski) [inline spec/controllers/posts_controller_spec.rb]:
```
  describe "POST #create" do
     let(:post_params) { attributes_for(:post) }

     subject do
       sign_in(user)
       post :create, params: { post: post_params }
     end

      it 'requires authentication' do
        post :create
        expect(response).to redirect_to(new_user_session_path)
      end

      it "should success create" do
        subject
        expect(response).to have_http_status(:found)
      end

      it "should create new post in db" do
        expect{ subject }.to change(Post, :count).by(1)
      end
 
      it "will set notice" do
        subject
        expect(flash[:notice]).to be_present
      end
 
      context "when invalid" do
        let(:post_params) { attributes_for(:post, title: '') }
 
        it "will render new template" do
          subject
          expect(response).to render_template(:new)
        end
      end
    end
```

File: spec/controllers/posts_controller_spec.rb [hunk 3]
```diff
@@ -20,16 +23,142 @@
...
+
+  describe "DELETE 'destroy'" do
+    let(:other_user) { create(:user) }
+    let(:post) { create(:post, user: user) }
+    let(:post_params) { attributes_for(:post) }
+
+    subject do
+      delete :destroy, params: { id: post.id, post: post_params }
+    end
+
```
Comment (mosinski) [inline spec/controllers/posts_controller_spec.rb]: Always ensure that it requires authentication when should be:
```
     it 'requires authentication' do
       delete :destroy
       expect(response).to redirect_to(new_user_session_path)
     end
```

File: app/mailers/subscriber_mailer.rb [hunk 4]
```diff
@@ -2,6 +2,6 @@ class SubscriberMailer < ApplicationMailer
 	def new_post(subscriber, post)
```
Comment (Deathis) [inline app/mailers/subscriber_mailer.rb]: Parametr `post` nie jest użyty.

File: spec/controllers/posts_controller_spec.rb [hunk 5]
```diff
@@ -20,16 +23,161 @@
...
+      expect(response).to redirect_to post_path
+    end
+
+    context "when user is an author" do
+
+      it "updates requested post" do
+        sign_in(user)
+        subject
+        post.reload
+        expect(post.title).to eq('test')
```
Comment (mosinski) [inline spec/controllers/posts_controller_spec.rb]: Shorter:
```
  expect(post.reload.title).to eq('test')
```

File: config/routes.rb [hunk 6]
```diff
@@ -1,6 +1,5 @@
 Rails.application.routes.draw do
   get 'settings/index'
-  get 'subscribers/index'
   get 'password_resets/new'
   get 'project/index'
   get 'projects/index'
```
Comment (mosinski) [inline config/routes.rb]: I quess only one of the route is needed here

**Pull request merged**